quinn-proto/src/tests.rs

```diff
@@ -236,11 +252,12 @@ struct TestEndpoint {
     addr: SocketAddr,
     socket: Option<UdpSocket>,
     timers: [Option<Instant>; Timer::COUNT],
-    conn: Option<ConnectionHandle>,
     outbound: VecDeque<Transmit>,
     delayed: VecDeque<Transmit>,
     inbound: VecDeque<(Instant, Option<EcnCodepoint>, Box<[u8]>)>,
     accepted: Option<ConnectionHandle>,
+    connections: FnvHashMap<ConnectionHandle, Connection>,
+    conn_events: FnvHashMap<ConnectionHandle, VecDeque<ConnectionEvent>>,
```

Comment:
I see it more as something that helps test for correctness by requiring the use of the correct `ConnectionHandle`s. I don't think it would simplify all that much to use a single `Option<Connection>` per `TestEndpoint` or some such.

Reply:
The main thing would be not having to thread the `ConnectionHandle` through to `client_conn_mut` and such. Not that significant, though, and we probably really should get some tests for multiple concurrent connections in one endpoint sooner or later.